pyccl/halos/halo_model.py

```diff
@@ -147,15 +149,17 @@ def profile_norm(self, cosmo, a, prof):
             float or array_like: integral value.
         """
         self._get_ingredients(cosmo, a, get_bf=False)
-        uk0 = prof.fourier(cosmo, self.precision['k_norm'],
+        uk0 = prof.fourier(cosmo, self._prec['k_norm'],
                            self._mass, a, mass_def=self.mass_def).T
-        return 1 / self._integrate_over_mf(uk0)
+        return 1. / self._integrate_over_mf(uk0)
 
     def get_profile_norm(self, cosmo, a, prof):
-        """Compute ``profile_norm`` if ``prof.normprof`` is ``True``."""
-        if prof.normprof:
-            return self.profile_norm(cosmo, a, prof)
-        return np.ones_like(a)[()]
+        """Compute the normalization of a profile."""
+        self._get_ingredients(cosmo, a, get_bf=False)
+        uk0 = prof._normalization(self)(cosmo=cosmo, a=a)
+        if isinstance(uk0, (int, float)):
+            return uk0
```

Comment:
if there's an integral we return 1/integral, so let's assume that if it's a number we should do 1/uk0

Reply:
ah yes, I was only thinking about the norm=1 case